xwiki-commons-core/xwiki-commons-diff/xwiki-commons-diff-display/src/main/java/org/xwiki/diff/display/internal/DefaultUnifiedDiffDisplayer.java

```diff
@@ -500,7 +500,9 @@ public void setLastConflict(Conflict<E> lastConflict)
     private <E, F> List<UnifiedDiffElement<E, F>> getUnmodifiedElements(List<E> previous, int start, int end)
     {
         List<UnifiedDiffElement<E, F>> unmodifiedElements = new ArrayList<>();
-        for (int i = start; i < end; i++) {
+        // Prevent exceeding the size of the previous list in the loop.
+        int realEnd = Math.min(end, previous.size());
+        for (int i = start; i < realEnd; i++) {
```

Comment:
What if ``start`` is negative? I don't think this method should validate the start and end points. Again, I'm worried that you're going to hide a real problem that is elsewhere.